File: code-test/solutions/python/day_006.py
from collections import deque

def solution(priorities, location):
    # TODO: location 위치의 문서가 몇 번째로 출력되는지 반환하세요.
    # 예: priorities = [2, 1, 3, 2], location = 2 -> 1

    queue = deque()
    answer = 0

    for i in range(0, len(priorities)) :
        queue.append([priorities[i], i])

    while (queue) :
        current = queue.popleft()
        currentPriorities = current[0]
        currentIndex = current[1]

        big = False

        for item in queue :
            if (item[0] > currentPriorities) :
                big = True
                break

        if (big) :
            queue.append(current)
        else :
            answer+=1
            if (location == currentIndex):
                return answer


    return answer

File: code-test/solutions/java/Day006.java
import java.util.*;

public class Day006 {
    public static int solution(int[] priorities, int location) {
        // TODO: location 위치의 문서가 몇 번째로 출력되는지 반환하세요.
        // 예: priorities = [2, 1, 3, 2], location = 2 -> 1
        
        Queue<int[]> queue = new LinkedList<>();
        int answer = 0;

        for(int i=0; i<priorities.length; i++) {
            queue.offer(new int[]{priorities[i], i});
        }
        
        while (!queue.isEmpty()) {
            int[] current = queue.poll();
            int currentPriorities = current[0];
            int currentIndex = current[1];

            boolean big = false;

            for(int[] item : queue) {
                if (item[0] > currentPriorities) {
                    big = true;
                    break;
                }
            }

            if(big) {
                queue.offer(current);
            } else {
                answer++;

                if (currentIndex == location) {
                    return answer;
                }
            }
        }



        return answer;
    }
}

